diff --git a/lib/utils/validator-factory.js b/lib/utils/validator-factory.js
index fde823b7d7b3c180e9dfcb44236b6c776539321b..763e40f9dc33562ef3754d8e2e847945a0976450 100644
--- a/lib/utils/validator-factory.js
+++ b/lib/utils/validator-factory.js
@@ -6,6 +6,7 @@ Object.defineProperty(exports, "__esModule", { value: true });
 exports.compile = compile;
 const json_schema_migrate_x_1 = require("json-schema-migrate-x");
 const ajv_1 = __importDefault(require("./ajv"));
+const ajvErrors = require("ajv-errors");
 const schema_1 = require("./schema");
 const lazyRegExpEngine = (str, flags) => {
     let error;
@@ -31,6 +32,7 @@ const ajv = new ajv_1.default({
         regExp: lazyRegExpEngine,
     },
 });
+ajvErrors(ajv);
 ajv.addMetaSchema(require("ajv/lib/refs/json-schema-draft-06.json"));
 function unescapeFragment(str) {
     return unescapeJsonPointer(decodeURIComponent(str));
diff --git a/package.json b/package.json
index e161c6249abc7ea00e0c81fba54283930ef9e5d5..33428aa243d8155bb0547498865cb0dddbdf9703 100644
--- a/package.json
+++ b/package.json
@@ -24,7 +24,8 @@
     "json-schema"
   ],
   "peerDependencies": {
-    "eslint": ">=6.0.0"
+    "eslint": ">=6.0.0",
+    "ajv-errors": "*"
   },
   "dependencies": {
     "@eslint-community/eslint-utils": "^4.3.0",
