diff --git a/lib/resolveTaskFn.js b/lib/resolveTaskFn.js
index 471f39b22fcc5fb7d6820e2bc917baa496d92675..878189a4bb1b2f7ebf8a8cb6797076837f56f23a 100644
--- a/lib/resolveTaskFn.js
+++ b/lib/resolveTaskFn.js
@@ -11,7 +11,7 @@ import { TaskError } from './symbols.js'
 /**
  * @see https://github.com/sindresorhus/execa/blob/f4b8b3ab601c94d1503f1010822952758dcc6350/lib/command.js#L32-L37
  */
-const escapeSpaces = (input) => input.replaceAll(' ', '\\ ')
+const escapeSpaces = (input) => `'${input}'`;

 const TASK_ERROR = 'lint-staged:taskError'

@@ -154,7 +154,7 @@ export const resolveTaskFn = ({
     // Only use topLevelDir as CWD if we are using the git binary
     // e.g `npm` should run tasks in the actual CWD
     cwd: /^git(\.exe)?/i.test(cmd) ? topLevelDir : cwd,
-    preferLocal: true,
+    preferLocal: false,
     reject: false,
     shell,
     stdin: 'ignore',
